tests/rptest/redpanda_cloud_tests/high_throughput_test.py

```diff
@@ -244,6 +253,8 @@ def __init__(self, test_ctx: TestContext, *args, **kwargs):
                              disable_cloud_storage_diagnostics=True,
                              **kwargs)
 
+        assert isinstance(self.redpanda, RedpandaServiceCloud)
```

Comment:
Runtime check for our type assertion above (type checkers are not smart enough for this assert alone to be enough).